### PR TITLE
Use JWT authentication in Grader interactions

### LINKED ISSUE #29: Radar does not work with MOOC-Grader due to missing JWT auth implementation
Radar has not been updated for the new v1.12 JWT authentication in the A+ service protocols.

Can Radar be used at all anymore in this state? Test the Radar features more in production to see how broken they are.

Errors in the radar.cs logs:
InvalidHeader when requesting template contents from https://grader.cs.aalto.fi/exercise_template/traky2021-en/lisatehtavat_en_mergesort_mergesort/mergesort.py: Expected response content with MIME type text/plain but got content type text/html; charset=utf-8
Report
InvalidHeader when requesting template contents from https://grader.cs.aalto.fi/exercise_template/traky2021-en/lisatehtavat_en_mergesort_mergesort/mergesort.py: Expected response content with MIME type text/plain but got content type text/html; charset=utf-8

I didn't investigate this yet, but I suspect that Radar is getting the HTML page with the login form instead of the wanted exercise template file. Accessing the grader.cs API now requires JWT token, but Radar does not yet have any JWT implementation.

Maybe the library https://github.com/apluslms/aplus-auth/ is useful?

### PR DESCRIPTION
**Pull request by PasiSa**
Update the GET request for exercise templates from Grader to
use the aplus-auth library for JWT authentication. Related
APLUS_AUTH configuration also needs to be set in local_settings.py.
Fixes #29.

# Description

**What?**

Very small change, but making a pull request nevertheless, in case someone wants to check it out before merging.

**Why?**

[ANSWER HERE]

**How?**

[ANSWER HERE]

Fixes #<issue number>

# Testing

**Remember to add or update unit tests for new features and changes.**

* How to [test your changes in A-plus](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations)
* How to [test accessibility](https://wiki.aalto.fi/display/EDIT/How+to+check+the+accessibility+of+pull+requests)

**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [ ] Django unit tests.
- [ ] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [ ] Manual testing.

[ADD A DESCRIPTION ABOUT WHAT YOU TESTED MANUALLY]

**Did you test the changes in**

- [ ] Chrome
- [ ] Firefox
- [ ] This pull request cannot be tested in the browser.

**Think of what is affected by these changes and could become broken**

# Translation

- [ ] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [ ] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?
- [ ] Did you use Python type hinting in all functions that you added or edited? ([type hints](https://docs.python.org/3/library/typing.html) for function parameters and return values)

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

*Clean up your git commit history before submitting the pull request!*

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -10,3 +10,4 @@ git+https://github.com/apluslms/a-plus-client.git@v1.1.0#egg=a_plus_client
 python-memcached >= 1.59, < 2
 git+https://github.com/apluslms/greedy-string-tiling.git@v0.12.0
 django-debug-toolbar >= 3.2
+aplus-auth~=0.2.0
```
Comment (Mankro) [inline requirements.txt:13]: This file already used the cleaner coding style that uses whitespace around the operators.

**Review by Mankro — CHANGES_REQUESTED** [review]
Good! Though, you forgot the whitespace around the operator in requirements.txt.

Just a thought:
I wonder if the template should be fetched from the grader or gitmanager nowadays.

File: requirements.txt (hunk 1 above)
Comment (PasiSa) [reply requirements.txt]: Fixed

**Comment by PasiSa** [comment]
> Just a thought: I wonder if the template should be fetched from the grader or gitmanager nowadays.

Probably yes. Also Tuomas had some further ideas how the template fetching should perhaps be redesigned in future, but I thought that it is better to first do a quick fix for the immediate problem, in case someone needs radar soon, and think about the other possible changes as a separate step later.

**Review by Mankro — APPROVED** [review]

**Pull request merged**